code_generation/producers/ditaupairquantities.py

```diff
@@ -0,0 +1,179 @@
+import code_generation.quantities as q
+from code_generation.producer import Producer, ProducerGroup
+
+####################
+# Set of general producers for DiTauPair Quantities
+####################
+
+pt_1 = Producer(
+    name="pt_1",
+    call="quantities::pt({df}, {output}, {input})",
+    input=[q.p4_1],
+    output=[q.pt_1],
+    scopes=["mt", "et", "tt", "em"],
+)
+pt_2 = Producer(
+    name="pt_2",
+    call="quantities::pt({df}, {output}, {input})",
+    input=[q.p4_2],
+    output=[q.pt_2],
+    scopes=["mt", "et", "tt", "em"],
+)
+eta_1 = Producer(
+    name="eta_1",
+    call="quantities::eta({df}, {output}, {input})",
+    input=[q.p4_1],
+    output=[q.eta_1],
+    scopes=["mt", "et", "tt", "em"],
+)
+eta_2 = Producer(
+    name="eta_2",
+    call="quantities::eta({df}, {output}, {input})",
+    input=[q.p4_2],
+    output=[q.eta_2],
+    scopes=["mt", "et", "tt", "em"],
+)
+phi_1 = Producer(
+    name="phi_1",
+    call="quantities::phi({df}, {output}, {input})",
+    input=[q.p4_1],
+    output=[q.phi_1],
+    scopes=["mt", "et", "tt", "em"],
+)
+phi_2 = Producer(
+    name="phi_2",
+    call="quantities::phi({df}, {output}, {input})",
+    input=[q.p4_2],
+    output=[q.phi_2],
+    scopes=["mt", "et", "tt", "em"],
+)
+m_vis = Producer(
+    name="m_vis",
+    call="quantities::m_vis({df}, {output}, {input_vec})",
+    input=[q.p4_1, q.p4_2],
+    output=[q.m_vis],
+    scopes=["mt", "et", "tt", "em"],
+)
+####################
+# Set of channel specific producers
+####################
+mass_1 = Producer(
```

Comment:
this naming will lead to collisions when other channels are intoduced with a corresponding quantity. Can't we take it from the 4-Vector, so it could be the same definition for all channels?